lib/semantics/check-omp-structure.cc

```diff
@@ -246,7 +294,47 @@ void OmpStructureChecker::Enter(const parser::OmpProcBindClause &) {
 void OmpStructureChecker::Enter(const parser::OmpReductionClause &) {
   CheckAllowed(OmpClause::REDUCTION);
 }
-void OmpStructureChecker::Enter(const parser::OmpScheduleClause &) {
+
+bool OmpStructureChecker::ScheduleModifierHasType(
+    const parser::OmpScheduleClause &x,
+    const parser::OmpScheduleModifierType::ModType &type) {
+  const auto &modifier{std::get<0>(x.t)};
+  if (modifier.has_value()) {
+    const auto &modType1{std::get<0>(modifier->t)};
+    const auto &modType2{std::get<1>(modifier->t)};
+    if ((modType1.v).v == type ||
+        (modType2.has_value() && (modType2->v).v == type)) {
+      return true;
+    }
+  }
+  return false;
+}
+void OmpStructureChecker::Enter(const parser::OmpScheduleClause &x) {
   CheckAllowed(OmpClause::SCHEDULE);
+
+  // 2.7 Loop Construct Restriction
+  if (GetContext().directive == OmpDirective::DO) {
+    const auto &kind{std::get<1>(x.t)};
+    const auto &chunk{std::get<2>(x.t)};  // chunk_size
+    if ((kind == parser::OmpScheduleClause::ScheduleType::Runtime ||
+            kind == parser::OmpScheduleClause::ScheduleType::Auto) &&
+        chunk.has_value()) {
+      context_.Say(GetContext().clauseSource,
+          "When SCHEDULE(%s) is specified, "
+          "CHUNK_SIZE must not be specified"_err_en_US,
```

Comment:
I suggest "When SCHEDULE has %s specified it must not have chunk size specified".
CHUNK_SIZE is not something that appears in the source like SCHEDULE and AUTO.

Reply:
Addressed in commit [1b57193](https://github.com/flang-compiler/f18/pull/536/commits/1b57193d2377c5121f16075950b64ea77fa72e2e).